scipy/misc/common.py

```diff
@@ -97,6 +97,7 @@ def derivative(func, x0, dx=1.0, n=1, args=(), order=3):
     Notes
     -----
     Decreasing the step size too small can result in round-off error.
+    Behavior at a non-differentiable point is unreliable.
```

Comment:
```suggestion
    This function assumes that ``func`` is differentiable at ``x0``. 
    If ``func`` is not differentiable at ``x0``, the result is undefined.
```

Reply:
The behavior is quite well defined, though.

Reply:
Is it, considering we don't know what the user's `func` is going to do near `x0`?

Reply:
Central differences still computes a well-defined result there.

Reply:
Now that it is indeed documented to compute a central difference, how about we say that "This function assumes that ``func`` is differentiable at ``x0`` and does not check if it really is" or "computes the central difference assuming that `func` is differentale at `x0`" and leave it at that.

Reply:
That would be valid, but I don't think that would have helped gh-12887 much. Probably you would want to say something that central differences gives more accurate results when the requested derivative doesn't change much over the range `(x0-dx, x0+dx)` (or wider for larger orders). This includes when the requested derivative is discontinuous in that range.

Reply:
How about "If the points x-dx and x+dx span a point where the true derivative is discontinuous, then the gradient calculated by this central difference will be different to a gradient calculated with either forwards or backwards differences, and is unlikely to be a good approximation of the true derivative at x"
Is it a bit long winded?